src/sidebar/store/modules/toast-messages.js

```diff
@@ -0,0 +1,71 @@
+import * as util from '../util';
+
+/**
+ * A store module for managing a collection of toast messages. This module
+ * maintains state only; it's up to other layers to handle the management
+ * and interactions with these messages.
+ */
+
+function init() {
+  return {
+    messages: [],
+  };
+}
+
+const update = {
+  ADD_MESSAGE: function(state, action) {
+    return {
+      messages: state.messages.concat([Object.assign({}, action.message)]),
```

Comment:
I think we can simplify this down to just
`state.messages.concat({...action.message})`

that copies the the message object and concat should treat that as a single item added to a new array.